acceptancetests/assess_persistent_storage.py

```diff
@@ -165,7 +165,7 @@ def assert_app_removal_msg(client, charm_name):
     # merge_stderr=True is required
     app_removal_output = client.get_juju_output(
         'remove-application', charm_name, '--show-log',
-        include_e=False, merge_stderr=True)
+        include_e=False, merge_stderr=True).decode('utf-8')
```

Comment:
Maybe put `.decode('utf-8')` into the `get_juju_output` method?

Reply:
In some cases we pipe that to `json.loads`. It's easier to patch at the call site where we need to run a regex against that.
I think we should just make things work under python3 and slap a big "caveat emptor: use the bash framework for writing new tests" warning in the readme.

Reply:
Fine :+1:

Reply:
Hmmm... ok, let me change that everywhere :-(